app/models/rbac.rb

```diff
@@ -161,17 +171,13 @@ def self.combine_filtered_ids(u_filtered_ids, b_filtered_ids, m_filtered_ids, d_
   end
 
   def self.find_targets_with_indirect_rbac(klass, scope, rbac_filters, find_options = {}, user_or_group = nil)
-    parent_class   = rbac_class(klass)
-    u_filtered_ids = get_self_service_object_ids(user_or_group, parent_class)
-    b_filtered_ids = get_belongsto_filter_object_ids(parent_class, rbac_filters['belongsto'])
-    m_filtered_ids = get_managed_filter_object_ids(parent_class, parent_class, rbac_filters['managed'])
-    d_filtered_ids = rbac_filters['ids_via_descendants']
-    filtered_ids   = combine_filtered_ids(u_filtered_ids, b_filtered_ids, m_filtered_ids, d_filtered_ids)
+    parent_class = rbac_class(klass)
+    filtered_ids = calc_filtered_ids(parent_class, parent_class, rbac_filters, user_or_group).first
```

Comment:
Since the method is returning two values, I think prefer an underscorized variable over .first.  This is because from a cursory reading of the code, .first implies an Array of things, as opposed to 2 return values that happen to be in an Array.

So something like...

```
filtered_ids, _u_filtered_ids = calc_filtered_ids(parent_class, parent_class, rbac_filters, user_or_group)

# or just

filtered_ids, _ = calc_filtered_ids(parent_class, parent_class, rbac_filters, user_or_group)
```